app/validators/array_inclusion_validator.rb

```diff
@@ -0,0 +1,8 @@
+class ArrayInclusionValidator < ActiveModel::EachValidator
```

Comment:
Shouldn't the `validates_inclusion_of` validator already enable us to perform this exact validation?

Reply:
AFAIU `validates_inclusion_of` works only if the given attribute is non an enumerable.

Given a Player with many technologies and a `validates_inclusion_of` validation:

```ruby
class Player < ApplicationRecord
  validates :technologies, inclusion: { in: Technology.all }
end
```

… the [InclusionValidator](https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/activemodel/lib/active_model/validations/inclusion.rb#L11) would attempt to call:

```ruby
technologies = Technology.all.slice(0, 1)
=> ['<Technology: {"key"=>"antimass_deflectors", "title"=>"Antimass Deflectors", "requires"=>nil, "color"=>"B"}>']
Technology.all.include?(technologies)
=> false
```

While the `ArrayInclusionValidator` checks if all items in a given attribute-Array are also part of the provided Array for comparison:

```ruby
technologies = Technology.all.slice(0, 1)
=> ['<Technology: {"key"=>"antimass_deflectors", "title"=>"Antimass Deflectors", "requires"=>nil, "color"=>"B"}>']
(technologies - Technology.all).blank?
=> true
```

Reading through the implementation of [`ActiveModel::Validations::Clusivity#inclusion_method `](https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/activemodel/lib/active_model/validations/clusivity.rb#L40-L50) I think we _could_ rely on `validates_inclusion_of` if we'd implement `include?` on `ValueObject`.

```diff
diff --git a/app/models/player.rb b/app/models/player.rb
index f416855..e79987f 100644
--- a/app/models/player.rb
+++ b/app/models/player.rb
@@ -10,7 +10,7 @@ class Player < ApplicationRecord
   validates :seat_number, presence: true
 
   attribute :technologies, :value_object, default: []
-  validates :technologies, array_inclusion: { in: Technology.all }
+  validates :technologies, inclusion: { in: Technology }
 
   before_create :add_starting_technologies
 
diff --git a/app/models/value_object.rb b/app/models/value_object.rb
index 4bb377a..e9ed061 100644
--- a/app/models/value_object.rb
+++ b/app/models/value_object.rb
@@ -15,6 +15,11 @@ class ValueObject
     end
   end
 
+  def self.include?(objects)
+    objects = Array(objects)
+    (objects.uniq - all).blank?
+  end
+
   def self.find_by_key(key)
     all.detect { |obj| obj.key.to_s == key.to_s }
   end
```

which effectively allows us to call.

```ruby
technologies = Technology.all.slice(0, 1)
Technology.include?(technologies)
=> true
```

For me the `array_inclusion`-Validator is a bit easier to understand since one does not have to know about the existence of [`ActiveModel::Validations::Clusivity#inclusion_method `](https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/activemodel/lib/active_model/validations/clusivity.rb#L40-L50) while the alternative is also elegant

Reply:
What do you think? :)

Reply:
Good point, didn't thought about that 🤔 Let's keep the `array_inclusion` validator then 👍